### ISSUE TITLE
OpenMP CUDA-offload regression: Unable to find 'nvlink' in path.

### ISSUE THREAD

**Issue by Meinersbur** [llvm/llvm-project#53573]
The CUDA offload buildbot fails after giant push b8239af0eeedfa67d9fac28d20685b7a09e2935a to 4d4587d5b00028fc525a2296914463ca70fbbf76.

http://meinersbur.de:8011/#/builders/11/builds/1046

```
******************** TEST 'libomptarget :: nvptx64-nvidia-cuda-newDriver :: offloading/bug49021.cpp' FAILED ********************
Script:
--
: 'RUN: at line 1';   /home/buildbot-worker/minipc-1050ti-linux/rundir/llvm.build/./bin/clang++ -fopenmp -pthread -fno-experimental-isel --cuda-path=/opt/cuda  -I /home/buildbot-worker/minipc-1050ti-linux/rundir/llvm.src/openmp/libomptarget/test -I /home/buildbot-worker/minipc-1050ti-linux/rundir/llvm.build/projects/openmp/runtime/src -L /home/buildbot-worker/minipc-1050ti-linux/rundir/llvm.build/projects/openmp/libomptarget -L /home/buildbot-worker/minipc-1050ti-linux/rundir/llvm.build/lib  -fno-openmp-implicit-rpath -Wl,-rpath,/home/buildbot-worker/minipc-1050ti-linux/rundir/llvm.build/projects/openmp/libomptarget -Wl,-rpath,/home/buildbot-worker/minipc-1050ti-linux/rundir/llvm.build/lib -Wl,-rpath,/opt/cuda/lib64 --libomptarget-nvptx-bc-path=/home/buildbot-worker/minipc-1050ti-linux/rundir/llvm.build/projects/openmp/libomptarget -fopenmp-new-driver -fopenmp-targets=nvptx64-nvidia-cuda /home/buildbot-worker/src/llvm-project/openmp/libomptarget/test/offloading/bug49021.cpp -o /home/buildbot-worker/minipc-1050ti-linux/openmp-offload-cuda-project/llvm.build/projects/openmp/libomptarget/test/nvptx64-nvidia-cuda-newDriver/offloading/Output/bug49021.cpp.tmp -O3 && /home/buildbot-worker/minipc-1050ti-linux/openmp-offload-cuda-project/llvm.build/projects/openmp/libomptarget/test/nvptx64-nvidia-cuda-newDriver/offloading/Output/bug49021.cpp.tmp
--
Exit Code: 1
Command Output (stdout):
--
$ ":" "RUN: at line 1"
note: command had no output on stdout or stderr
$ "/home/buildbot-worker/minipc-1050ti-linux/rundir/llvm.build/./bin/clang++" "-fopenmp" "-pthread" "-fno-experimental-isel" "--cuda-path=/opt/cuda" "-I" "/home/buildbot-worker/minipc-1050ti-linux/rundir/llvm.src/openmp/libomptarget/test" "-I" "/home/buildbot-worker/minipc-1050ti-linux/rundir/llvm.build/projects/openmp/runtime/src" "-L" "/home/buildbot-worker/minipc-1050ti-linux/rundir/llvm.build/projects/openmp/libomptarget" "-L" "/home/buildbot-worker/minipc-1050ti-linux/rundir/llvm.build/lib" "-fno-openmp-implicit-rpath" "-Wl,-rpath,/home/buildbot-worker/minipc-1050ti-linux/rundir/llvm.build/projects/openmp/libomptarget" "-Wl,-rpath,/home/buildbot-worker/minipc-1050ti-linux/rundir/llvm.build/lib" "-Wl,-rpath,/opt/cuda/lib64" "--libomptarget-nvptx-bc-path=/home/buildbot-worker/minipc-1050ti-linux/rundir/llvm.build/projects/openmp/libomptarget" "-fopenmp-new-driver" "-fopenmp-targets=nvptx64-nvidia-cuda" "/home/buildbot-worker/src/llvm-project/openmp/libomptarget/test/offloading/bug49021.cpp" "-o" "/home/buildbot-worker/minipc-1050ti-linux/openmp-offload-cuda-project/llvm.build/projects/openmp/libomptarget/test/nvptx64-nvidia-cuda-newDriver/offloading/Output/bug49021.cpp.tmp" "-O3"
# command stderr:
/home/buildbot-worker/minipc-1050ti-linux/rundir/llvm.build/./bin/clang-linker-wrapper: error: Unable to find 'nvlink' in path
clang-15: error: linker command failed with exit code 1 (use -v to see invocation)
error: command failed with exit status: 1
--
********************
```

This is likely due to not taking the `--cuda-path` into account when searching for `nvlink`.

**Comment by jhuber6** [comment]
Not accepting --cuda-path is definitely the issue. I'll need to copy the behaviour of `CudaInstallationDetector` into the tool and forward `--cuda-path` as well. I'll have a fix up soon.

**Comment by shiltian** [comment]
You might not need to copy the code to the wrapper. Just pass that information to the wrapper probably via `--cuda-path`.

**Comment by jhuber6** [comment]
> You might not need to copy the code to the wrapper. Just pass that information to the wrapper probably via `--cuda-path`.

In the process of doing that, just forwarding what we get from the installation detector to the tool. I don't think I'll need to copy the same functionality for ROCm since we only use `lld` for AMDGPU linking. Right now we simply look at where the tool itself is installed for things like `lld`, not sure if that's sufficient.

**Comment by jhuber6** [comment]
https://reviews.llvm.org/D118944

**Comment by tstellar** [comment]
Merged: 4db1b3bdf13d23be5247a7b1222176a00f774288
